### PR TITLE
Trim down feature tests for file modes

### PR DESCRIPTION
**Pull request by siteshwar**

### CODE REVIEW

File: src/lib/libast/features/mode.c [hunk 1]
```diff
@@ -27,192 +27,22 @@
...
-	int	n;
-	int	idperm;
-	int	idtype;
-
-	idperm = idtype = 1;
 #ifndef S_ITYPE
 #ifdef	S_IFMT
 	printf("#define S_ITYPE(m)	((m)&S_IFMT)\n");
 #else
 	printf("#define S_ITYPE(m)	((m)&~S_IPERM)\n");
```
Comment (krader1961) [inline src/lib/libast/features/mode.c]: Ugh! I've never heard of the `S_IPERM` or `S_ITYPE` symbol. It's not defined on macOS or Linux. I can't find them mentioned in the APUE book either. So either this feature test can be completely eliminated or there are platforms we care about which have this unusual configuration. If the latter a comment to that effect would be a good idea.

Comment (siteshwar) [reply src/lib/libast/features/mode.c]: None of these macros are provided by default and there is some ast code that depend on them. I plan to clean it up later.

Comment (krader1961) [reply src/lib/libast/features/mode.c]: SGTM, ship it.

Comment (jelmd) [reply src/lib/libast/features/mode.c]: Ehmm, if it is not ready, it should never be shipped/commited! Wondering, where you learned to do such insane things ...

Comment (krader1961) [reply src/lib/libast/features/mode.c]: @jelmd, Your comments are not constructive and their tone is condescending. AFAICT you haven't actually contributed any changes. On the contrary, you've made it clear you won't be contributing. In issue #40 I suggested that you or @marcastel could contribute by helping port some of the Solaris patches to ksh93 to this project. You snidely replied:

> It was just a hint, to possibly safe you some time. If you prefer troubleshooting it by your own and provide your own fix, just go ahead.

So the next time you start to write a comment might think about what I've said if you want anyone to pay attention to what you have to say.

**Comment by jelmd** [comment]
@krader1961: Not sure, how you come to this conclusion. I've already shown, how one can build ksh only, provided related files and made suggestions, how to improve things, how things should/could be done and also mentioned, what IMHO should  not be done. I also gave a hint, where more patches are, which might help to get problems solved. You explained, why one should not use them/it is not worth to look at it, and now you demand, that other people should do it. I think, something doesn't match here.

**Pull request merged**